### DISCUSSION TITLE
OME and Python

### DISCUSSION THREAD

**Discussion by adolfusadams** [OvenMediaLabs/OvenMediaEngine#1612] [Help]
Can Tensorflow object detections be sent to OME from a Python script so that the object detections can be viewed in a live and vod stream?

**Comment by bchah** [comment]
I think this is something you would do with OvenPlayer rather than OME itself. There are many ways you could feed the video to your AI application from OME for analysis, and I imagine you'd want to send those results back to whichever application is hosting the OME playback with your own logic for drawing rectangles or listing the results to viewers on the web.

**Reply by adolfusadams** [reply]
Yes, I would want to send those results back to the application that is hosting the OME playback with my own logic. So as per your suggestion of using the OvenPlayer, I am not sure that would be the best approach.  I would need an effective approach that allows for native or direct Python communication with OME.

**Reply by bchah** [reply]
External communication with OME is performed using the REST API, which Python and many other languages / environments support. But there is no REST endpoint for simply sending ancillary data and making it available via the API - that's outside the scope of what OME does (and a potential security issue too).

Can you clarify what you expect OME would do with your AI analysis data?

**Reply by adolfusadams** [reply]
I expect OME to display all detected objects with their respective bounding boxes and labels in both live and vod streams. I also would want OME to record the video with the detections.

**Comment by naanlizard** [comment]
Sounds like you should send the source to your python script directly, or
to ome and then pipe it to your script, then have your script push a stream
to ome with the changed video you want

**Reply by adolfusadams** [reply]
Yes, based on my initial question, I would want to send the source to the Python script and then on to OME. So the question still is, can this be done with Python and OME regarding the given use case?

**Comment by naanlizard** [comment]
I don't see why not, but that's all stuff you have to program yourself. Ome
provides an API and will gladly ingest streams, but past that's it's all you

**Comment by getroot** [comment]
Simply put, you can push stream from OvenMediaEngine to your Python App over RTMP, and work on something in your Python App, and then push back to OvenMediaEngine over RTMP.

**Reply by adolfusadams** [reply]
My colleague and I have tried unsuccessfully to get object detections from a Python script to display on video in OME. Firstly, we would like to know if it is in fact definitely possible to display object detections directly on live and on-demand video from Python in OME. If the answer is yes, then we would be willing to pay for support or consultancy from OME to assist us in getting this to work. Kindly indicate if the answer is yes or no to the above and, if yes, then indicate how we can proceed.

**Reply by naanlizard** [reply]
You seem  extremely confused about what you're doing

Ome will display whatever video you feed it.  that's all it does. If you want an overlay you need to either make a separate player/app that takes in the video from ome and processes it in real time at the player end, or take the video from ome, change the video, and push the new video as a new stream

**Reply by adolfusadams** [reply]
We are definitely not confused about what we are doing. Have you ever used an object detection model with Tensorflow and OpenCV? It does not work like you mentioned.

**Reply by naanlizard** [reply]
Then perhaps you are explaining it poorly. If you generate a video stream, ome can ingest it and distribute it. If you want to do processing of a stream, ome will serve the video stream. 

I have used opencv but I'm still confused what exactly you're trying to do and why you're having trouble
